src/persistence.rs

```diff
@@ -195,10 +220,7 @@ impl<'a, L: Ledger, Meta: Send + Serialize + DeserializeOwned> AtomicKeystoreSto
         })()
         .await
         {
-            Ok(()) => {
-                self.commit().await;
-                Ok(())
-            }
+            Ok(()) => Ok(()),
```

Comment:
Good question. In general, calling `commit()` implicitly in resource modifying functions is bad -- the top level which synchronizes the different resources and the atomic store needs to know when one resource is being committed, because all the other resources need either a `commit_version` and `skip_version` at the same time, and we also need a `commit_version` on the overall atomic store. Therefore, resources should only call `commit` or `commit_version` when explicitly requested, in their public `commit` function.

`revert` does not have the same restriction. It merely undoes pending changes in a particular resource, it doesn't cause any version (local or global) to be advanced. Still, it's probably not needed here, because the top level transaction mechanism will call `revert` if we return an error from here. I will try getting rid of the revert just for conceptual symmetry, and see if the tests pass.

Reply:
Sounds good. Thanks for the explanation!